backend/src/worker_flow.rs

```diff
@@ -647,8 +647,11 @@ async fn push_next_flow_job(
             }
             JobPayload::Code(raw_code)
         }
-        FlowModuleValue::ForloopFlow { value, .. } => JobPayload::RawFlow {
-            value: (**value).clone(),
+        FlowModuleValue::ForloopFlow { modules, .. } => JobPayload::RawFlow {
+            value: FlowValue {
+                modules: (**modules).clone(),
+                failure_module: flow.failure_module.clone(),
```

Comment:
I hope this is a reasonable thing to do. I'm less certain about this being a good idea with the retry stuff but we can worry about that if it comes up later.